### DISCUSSION TITLE
在 docker 部署中使用宿主機系統代理

### DISCUSSION THREAD

**Discussion by QIN2DIM** [lobehub/lobehub#1254] [Self Hosting] [answered]
无法使用 OPENAI 和 GEMINI PRO 服务，在 docker-compose.yaml 中配置了 HTTPS_PROXY 和 HTTP_PROXY 指向宿主机的 hysteria2 监听的 HOST，但不起作用。对于 OPENAI，我配置了 OPENAI_PROXY_URL 指向了 cloudflare 的服务，间接奏效了，但 GEMINI PRO 就依然不能用。

```json
{
  "error": {
    "message": "[GoogleGenerativeAI Error]: Error fetching from https://generativelanguage.googleapis.com/v1/models/gemini-pro:streamGenerateContent?alt=sse: fetch failed"
  },
  "provider": "google"
}
```

OPENAI 和 GEMINI PRO 在检测可用性时，都是超时报错

![图片](https://github.com/lobehub/lobe-chat/assets/62018067/f2db0f84-1061-460e-aa79-86450238c2dc)

**Comment by arvinxx** [comment, marked as answer]
Google 的AI SDK 目前还不支持设定 `BASE_URL`: https://github.com/google/generative-ai-js/issues/25 。等待他们做了相应实现后才能在 LobeChat 侧集成。

建议可以考虑使用 Vercel 部署，可以绕过地区限制。

**Comment by QIN2DIM** [comment]
@arvinxx 感谢作者回复

实际上我想知道如何让容器内的 lobe-chat 服务使用宿主机已部署的系统代理，已部署的代理是可以正常使用的（绕过地区限制）。我在服务器上跑了 FastGPT，这个容器服务对外暴露了 [名为 AXIOS_PROXY 的环境变量](https://doc.fastgpt.in/docs/development/proxy/http_proxy/)，可以直接填写 proxy_host/proxy_port 就能使用宿主机代理了。

但对于 lobe-chat，我直接填写 `HTTPS_PROXY` 之类的环境变量似乎不起作用（不使用 iptables 透明代理之类的负责配置）。还是说 lobe-chat route 从编码层面就强制发起“直连”动作，不使用系统代理。

**Reply by arvinxx** [reply]
应该是后者。你如果看一下现有的dockerfile，应该能发现启动脚本就是 node index.js，这个指令没有带任何其他的代理配置。

所以执行以后请求应该都是从docker 容器层发出去的。

可能代理需要从这个思路上去考虑。

或者就是修改 dockerfile ，让其支持相应的代理配置。但这个我目前没有什么特别好的思路做无侵入性的实现。

**Comment by XYZliang** [comment]
我也正在寻找办法，不知道您解决没有？

**Comment by linth2005** [comment]
在宿主机配置透明代理 docker 使用host模式和宿主机同享网络即可 https://xtls.github.io/document/level-2/tproxy.html

**Comment by lihe6666** [comment]
对，我就是想在局域网用，什么vercel，不用，不用
